arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -468,28 +473,81 @@ std::pair<irs::string_ref, irs::string_ref> splitAnalyzerName( // split name
-///        dependencies are met
+/// @brief read analyzers from vocbase
+/// @return visitation compleated fully
 ////////////////////////////////////////////////////////////////////////////////
-arangodb::Result validateFeatures(irs::flags const& features) {
-  for(auto& feature: features) {
-    if (&irs::frequency::type() == feature) {
-      // no extra validation required
-    } else if (&irs::norm::type() == feature) {
-      // no extra validation required
-    } else if (&irs::position::type() == feature) {
-      if (!features.check(irs::frequency::type())) {
-        return arangodb::Result( // result
-          TRI_ERROR_BAD_PARAMETER, // code
-          std::string("missing feature '") + std::string(irs::frequency::type().name()) +"' required when '" + std::string(feature->name()) + "' feature is specified"
-        );
-      }
-    } else if (feature) {
+arangodb::Result visitAnalyzers( // visit analyzers
+  TRI_vocbase_t& vocbase, // vocbase to visit
+  std::function<arangodb::Result(arangodb::velocypack::Slice const& slice)> const& visitor // visitor
+) {
+  if (arangodb::ServerState::instance()->isClusterRole()) {
+    static const auto queryString = arangodb::aql::QueryString( // query to execute
+      std::string("FOR d IN ") + ANALYZER_COLLECTION_NAME + " RETURN d" // query
+    );
+    arangodb::aql::Query query( // query
+      false, vocbase, queryString, nullptr, nullptr, arangodb::aql::PART_MAIN // args
+    );
+    auto* queryRegistry = arangodb::QueryRegistryFeature::registry();
+    auto result = query.executeSync(queryRegistry);
+
+    if (TRI_ERROR_NO_ERROR != result.code) {
+      return arangodb::Result(result.code, result.details);
+    }
+
+    auto slice = result.result->slice();
+
+    if (!slice.isArray()) {
       return arangodb::Result( // result
-        TRI_ERROR_BAD_PARAMETER, // code
-        std::string("unsupported analyzer feature '") + std::string(feature->name()) + "'" // value
+        TRI_ERROR_INTERNAL, // code
+        std::string("failed to parse contents of collection '") + ANALYZER_COLLECTION_NAME + "' in database '" + vocbase.name() + " while visiting analyzers"
       );
     }
+
+    for (arangodb::velocypack::ArrayIterator itr(slice); itr.valid(); ++itr) {
+      auto res = visitor(itr.value().resolveExternal());
+
+      if (!res.ok()) {
+        return res;
+      }
+    }
+
+    return arangodb::Result();
+  }
+
+  arangodb::OperationOptions options;
+  arangodb::SingleCollectionTransaction trx( // transaction
+    arangodb::transaction::StandaloneContext::Create(vocbase), // context
+    ANALYZER_COLLECTION_NAME, // collection
+    arangodb::AccessMode::Type::READ // access more
+  );
+  auto res = trx.begin();
+
+  if (!res.ok()) {
+    return res;
+  }
+
+  auto commit  = irs::make_finally([&trx]()->void { trx.commit(); }); // end read-only transaction
+  auto result = trx.all(ANALYZER_COLLECTION_NAME, 0, 2, options);
```

Comment:
why are we reading at most 2 documents from the collection here?

Reply:
looks like a typo, but the code ignores that argument anyway, hence all the tests passed
changed to '0'